compiler/src/org.graalvm.compiler.core.common/src/org/graalvm/compiler/core/common/VectorDescription.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.graalvm.compiler.core.common;
+
+import org.graalvm.compiler.core.common.type.PrimitiveStamp;
+import org.graalvm.compiler.core.common.type.Stamp;
+
+public abstract class VectorDescription {
+
+    /**
+     * Returns the maximum number of elements with the given stamp that can be packed into a vector
+     * on the current architecture.
+     *
+     * @param stamp
+     * @return max vector width in elements
+     */
+    public final int maxVectorWidth(Stamp stamp) {
+        if (stamp instanceof PrimitiveStamp) {
+            return maxVectorWidth((PrimitiveStamp) stamp);
+        }
+
+        return 1;
+    }
+
+    protected abstract int maxVectorWidth(PrimitiveStamp stamp);
```

Comment:
We've created this class + method to determine the platform vector length. There could be a better way.